hubble/relay.go

```diff
@@ -265,7 +266,12 @@ disable-server-tls: true
 }
 
 func (k *K8sHubble) relayImage() string {
-	return utils.BuildImagePath(k.params.RelayImage, defaults.AgentImage, k.params.RelayVersion, defaults.Version)
+	image, err := utils.BuildImagePath(k.params.RelayImage, defaults.AgentImage, k.params.RelayVersion, defaults.Version)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
I think it may be a bit brittle to have this kind of "print and exit" logic sprinkled around these various functions, since the caller likely does not expect that calling into functions like `relayImage()` could cause the whole process to terminate. This could end up leaving the cluster in a weird partially-initialized state if the user passes a bad argument.

Ideally the validation of the version should occur very early in the CLI before the CLI takes any action in the cluster, then these "image path build" functions can just trust that they will successfully build a valid version.

Reply:
Do you mean place  the validation into for example: https://github.com/cilium/cilium-cli/blob/37d6bd6f4c75a9bf4e415b33a255c16d128c2ddf/internal/cli/cmd/install.go#L50 ? In this case CLI wouldn't even pass it if it fails.

Reply:
That's close. Digging from there through `NewK8sInstaller()` gets to `InstallParameters.validate()`, which sounds exactly like the right place for such logic:

https://github.com/cilium/cilium-cli/blob/37d6bd6f4c75a9bf4e415b33a255c16d128c2ddf/install/install.go#L1045